File: Examples/test-suite/php/cpp11_attribute_specifiers_runme.php
<?php

require "tests.php";

// New functions
check::functions(array('noReturn','noDiscard','noDiscardDeprecated','maybeUnused1','maybeUnused2','likely','test_string_literal'));
// New classes
check::classes(array('cpp11_attribute_specifiers','S'));
// New vars
check::globals(array('a', 'aa', 'b'));

check::equal(test_string_literal(), 'Test [[ and ]] in string literal', "test_string_literal() wrong");

check::done();

File: Examples/test-suite/lua/cpp11_attribute_specifiers_runme.lua
local v=require("cpp11_attribute_specifiers")

-- assert(not pcall(function() v.noReturn() end)) TODO crash instead of exception
assert(v.noDiscard())
assert(v.noDiscardDeprecated())
v.maybeUnused1(false)
-- v.maybeUnused1() TODO maybe_unused ignored
assert(v.maybeUnused2(true, false))
-- assert(not v.maybeUnused2(false)) TODO maybe_unused ignored
assert(v.likely(true, true))
assert(not v.likely(false, false))
-- assert(v.likely(true)) TODO maybe_unused ignored

assert(v.test_string_literal() == 'Test [[ and ]] in string literal', "test_string_literal")

local s = v.S()
assert(s ~= nil)

assert(v.aa == 0)
assert(swig_type(v.a) == "int *")
assert(v.b == 0)
